andela/knights-bn-backend
# #170947656 Send email verification link

#### What does this PR do?
- It sends an email verification link to the new user's email address. 
- Once the link is opened the user gets redirected back to the app.

#### Description of Task to be completed?
- Send a verification link to the newly signed up user's email address. 
- Save the verified user in the database immediately after verifying their account.

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` in your terminal
- run `cd knights-bn-backend` in your terminal
- run `git checkout  ft-verification-email-170947656`
- run `npm install` to install all packages. 
- run `npm run createTables` 
- run `npm run dev-start`
- copy this into your  **.env**  file: 
> `HOST_NAME: 'knights-bn-backend-staging.herokuapp.com'`

> `LOCAL_HOST='http://localhost:4000`'
- copy  the following in your Postman as a request body: 
```
{
     "firstName": "first_name",
     "lastName": "last_name",
     "gender": "Male",
    "passportNumber" :"1234567rv",
    "email": "your.email@example.com",
   "password": "password"
}
```
- Please use your valid email so you can be able to see the result after verifying your accound. 

- Hit the this endpoint: `http://localhost:4000/api/v1/auth/signup`

- Check your email address to see the message. 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947656](https://www.pivotaltracker.com/story/show/170947656)

#### Screenshots (if appropriate)
#### Postman' response body 

<img width="1440" alt="Screen Shot 2020-02-26 at 14 19 30" src="https://user-images.githubusercontent.com/25052255/75344310-0b41da00-58a3-11ea-9295-2e35384490e7.png">

#### My email address 
<img width="1440" alt="Screen Shot 2020-02-26 at 14 20 19" src="https://user-images.githubusercontent.com/25052255/75344364-2876a880-58a3-11ea-98c8-926fb28df94a.png">
